test/spec/modules/smartadserverBidAdapter_spec.js

```diff
@@ -1425,6 +1414,87 @@ describe('Smart bid adapter tests', function () {
       expect(bannerRequest).to.not.equal(null).and.to.not.be.undefined;
       expect(bannerRequest).to.have.property('bidfloor').and.to.equal(1.93);
     });
+
+    describe('#getBidFloor', () => {
+      let bid;
+      beforeEach(() => {
+        bid = {
+          mediaTypes: {
+            banner: {
+              sizes: [
+                [300, 250],
+                [300, 600]
+              ]
+            }
+          },
+          getFloor: (data) => {
+            if (data.mediaType === BANNER) {
+              if (data.size[0] === 300 && data.size[1] === 250) {
+                return { floor: 1.2 };
+              } else if (data.size[0] === 300 && data.size[1] === 600) {
+                return { floor: 1.4 };
+              } else if (data.size[0] === 30 && data.size[1] === 60) {
+                return 'string';
+              } else {
+                return { floor: 1.0 };
+              }
+            } else if (data.mediaType === VIDEO) {
+              if (data.size[0] === 640 && data.size[1] === 480) {
+                return { floor: 2.3 };
+              } else {
+                return { floor: 2.1 };
+              }
+            } else {
+              return {};
+            }
+          }
+        };
+      });
+
+      it('should return lowest floor from specified ones', () => {
+        expect(spec.getBidFloor(bid, 'DKK', BANNER)).to.deep.eq(1.2);
+      });
+
+      it('should return default floor for media type whatever size', () => {
+        bid.mediaTypes.banner.sizes.push([300, 400]);
+        expect(spec.getBidFloor(bid, 'DKK', BANNER)).to.deep.eq(1.0);
+      });
+
+      it('should return default floor', () => {
+        expect(spec.getBidFloor(bid, 'DKK', VIDEO)).to.deep.eq(0);
+      });
+
+      it('should return default floor when currency not passed', () => {
+        expect(spec.getBidFloor(bid, undefined, BANNER)).to.deep.eq(1.2);
+      });
+
+      it('should return handle not number from floor module', () => {
```

Comment:
Could you please rephrase it? It's not clear what is the handle in that case + the test check if floor is eq(0); Maybe it should be;
should return DEFAULT_FLOOR in case of not a number value from floor module ?

Reply:
Indeed, the name is a mess. This test tests line 345. I've pushed an update for the name. We take the DEFAULT_FLOOR in this case as one of the considered.